sidechain/consensus/aura/src/lib.rs

```diff
@@ -0,0 +1,314 @@
+//! Aura worker for the sidechain.
+//!
+//! It is inspired by parity's implementation but has ben greatly amended.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![cfg_attr(test, feature(assert_matches))]
+
+#[cfg(all(feature = "std", feature = "sgx"))]
+compile_error!("feature \"std\" and feature \"sgx\" cannot be enabled at the same time");
+
+#[cfg(all(not(feature = "std"), feature = "sgx"))]
+#[macro_use]
+extern crate sgx_tstd as std;
+
+use core::marker::PhantomData;
+use itp_storage_verifier::GetStorageVerified;
+use its_consensus_common::{Environment, Error as ConsensusError, Proposer};
+use its_consensus_slots::{SimpleSlotWorker, Slot, SlotInfo};
+use its_primitives::{
+	traits::{Block as SidechainBlockT, SignedBlock},
+	types::block::BlockHash,
+};
+use its_validateer_fetch::ValidateerFetch;
+use sp_runtime::{
+	app_crypto::{sp_core::H256, Pair, Public},
+	traits::Block as ParentchainBlock,
+};
+use std::{string::ToString, time::Duration, vec::Vec};
+
+mod verifier;
+
+pub use verifier::*;
+
+#[cfg(test)]
+mod mock;
+
+/// Aura consensus struct.
+pub struct Aura<AuthorityPair, ParentchainBlock, SidechainBlock, Environment, OcallApi> {
+	authority_pair: AuthorityPair,
+	ocall_api: OcallApi,
+	environment: Environment,
+	claim_strategy: SlotClaimStrategy,
+	_phantom: PhantomData<(AuthorityPair, ParentchainBlock, SidechainBlock)>,
+}
+
+impl<AuthorityPair, ParentchainBlock, SidechainBlock, Environment, OcallApi>
+	Aura<AuthorityPair, ParentchainBlock, SidechainBlock, Environment, OcallApi>
+{
+	pub fn new(
+		authority_pair: AuthorityPair,
+		ocall_api: OcallApi,
+		environment: Environment,
+	) -> Self {
+		Self {
+			authority_pair,
+			ocall_api,
+			environment,
+			claim_strategy: SlotClaimStrategy::Normal,
+			_phantom: Default::default(),
+		}
+	}
+
+	pub fn with_claim_strategy(mut self, claim_strategy: SlotClaimStrategy) -> Self {
+		self.claim_strategy = claim_strategy;
+
+		self
+	}
+}
+
+/// The fraction of total block time we are allowed to be producing the block. So that we have
+/// enough time send create and send the block to fellow validateers.
+pub const BLOCK_PROPOSAL_SLOT_PORTION: f32 = 0.8;
+
+#[derive(PartialEq, Eq, Debug)]
+pub enum SlotClaimStrategy {
+	/// try to produce a block always even if it's not the authors slot
+	/// Intended for first phase to see if aura production works
+	Always,
+	/// Only produce blocks, when it's the authors slot.
+	Normal,
```

Comment:
Just a minor issue: maybe we can think of a better name than 'normal'?

Reply:
Hmm, yes maybe. For me, normal implies `RoundRobin` we can change it to that to make it more clear 👍